src/Orleans.CodeGenerator/SerializerGenerator.cs

```diff
@@ -412,19 +410,22 @@ private static void AddSerializationMembers(ISerializableTypeDescription type, L
                     codecExpression = IdentifierName(instanceCodec.FieldName);
                 }
 
-                var expectedType = serializerFields.First(f => f is TypeFieldDescription tf && SymbolEqualityComparer.Default.Equals(tf.UnderlyingType, memberType));
-                var writeFieldExpr = ExpressionStatement(
-                        InvocationExpression(
-                            codecExpression.Member("WriteField"),
-                            ArgumentList(
-                                SeparatedList(
-                                    new[]
-                                    {
-                                        Argument(writerParam).WithRefOrOutKeyword(Token(SyntaxKind.RefKeyword)),
-                                        Argument(fieldIdDeltaExpr),
-                                        Argument(expectedType.FieldName.ToIdentifierName()),
-                                        Argument(member.GetGetter(instanceParam))
-                                    }))));
+               // When a static codec is available, we can call it directly and can skip passing the expected type, since it is known to be the static codec's field type:
```

Comment:
```suggestion
               // When a static codec is available, we can call it directly and can skip passing the expected type,
               // since it is known to be the static codec's field type:
```